File: backend/backend/src/main/java/com/designworks/backend/dto/request/TrabajoCreateRequest.java
package com.designworks.backend.dto.request;

import java.time.LocalDate;
import java.util.List;

import com.designworks.backend.entities.enums.Prioridad;

public record TrabajoCreateRequest(
        String titulo,
        String cliente,
        String descripcion,
        Prioridad prioridad,
        LocalDate fechaFin,
        List<ParticipanteRequest> participantes,
        List<RequisitoCreateRequest> requisitos
) {}

File: frontend/lib/features/trabajos/data/models/trabajo_create_request.dart
import 'enums.dart';

/// Equivalente Dart de TrabajoCreateRequest.java
class TrabajoCreateRequest {
  final String titulo;
  final String cliente;
  final String descripcion;
  final Prioridad prioridad;
  final String? fechaFin; // formato "yyyy-MM-dd"
  final List<Map<String, dynamic>> participantes;
  final List<Map<String, dynamic>> requisitos;

  TrabajoCreateRequest({
    required this.titulo,
    required this.cliente,
    required this.descripcion,
    required this.prioridad,
    this.fechaFin,
    this.participantes = const [],
    this.requisitos = const [],
  });

  Map<String, dynamic> toJson() => {
    'titulo': titulo,
    'cliente': cliente,
    'descripcion': descripcion,
    'prioridad': prioridad.toJson(),
    if (fechaFin != null) 'fechaFin': fechaFin,
    'participantes': participantes,
    'requisitos': requisitos,
  };
}